src/library/columncache.cpp

```diff
@@ -5,9 +5,13 @@
 #include "library/dao/cratedao.h"
 
 void ColumnCache::setColumns(const QStringList& columns) {
+    m_columnsByIndex.clear();
```

Comment:
it was better with a QMap since it wasn't possible for m_columnByIndex to be out of sync with m_columnIndexByEnum